codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttSubAckMessage.java

```diff
@@ -23,14 +23,21 @@
 
     public MqttSubAckMessage(
             MqttFixedHeader mqttFixedHeader,
-            MqttMessageIdVariableHeader variableHeader,
+            MqttMessageIdAndPropertiesVariableHeader variableHeader,
             MqttSubAckPayload payload) {
         super(mqttFixedHeader, variableHeader, payload);
     }
 
+    public MqttSubAckMessage(
+            MqttFixedHeader mqttFixedHeader,
+            MqttMessageIdVariableHeader variableHeader,
+            MqttSubAckPayload payload) {
+        this(mqttFixedHeader, variableHeader.withEmptyProperties(), payload);
+    }
+
     @Override
-    public MqttMessageIdVariableHeader variableHeader() {
-        return (MqttMessageIdVariableHeader) super.variableHeader();
+    public MqttMessageIdAndPropertiesVariableHeader variableHeader() {
```

Comment:
We can't do this as it would be API breakage

Reply:
`MqttMessageIdAndPropertiesVariableHeader` is a subclass of `MqttMessageIdVariableHeader` - the value returned from `variableHeader` here will match the expectations of the users of the coded.

Reply:
Unfortunally this will still not work as it will break binary combat as far as I know

Reply:
if you run `mvn clean package` it should check for these kind of things and fail the build

Reply:
Unfortunately, wasn't able to run this check today due to https://github.com/netty/netty/issues/10114 - the full build of the fresh Netty code from branch `4.1` (without my changes) didn't complete. Tomorrow I'll try to skip `epoll` transport build and see if it goes forward.

Reply:
Verified in `coded-mqtt` folder with `mvn clean verify`. You were right about the return types: https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html#jls-13.4.15. 
Fixed this and other compatibility issues detected by `japicmp`.